tests/api_test/api_test.cpp

```diff
@@ -96,9 +97,11 @@ _program_load_helper(
         return error;
     }
 
-    *program_fd = bpf_program__fd(program);
+    if (program != nullptr) {
+        *program_fd = bpf_program__fd(program);
+    }
     *object = new_object;
-    return 0;
+    return EBPF_SUCCESS;
```

Comment:
nit: This function returns `-errno`

Reply:
No, it's already inconsistent.  See line 83.  Making it consistently return ebpf_result_t or -errno is outside the scope of this PR.

Reply:
Agree that we should fix code at line 83. But I think this change is making it even more inconsistent.

Reply:
Updated as requested